### ISSUE TITLE
Error during Training in IncrementalEncoder.encode

### ISSUE THREAD

**Issue by nost99** [Deci-AI/super-gradients#1160]
### 💡 Your Question

I'm using the follwong versions:
Python 3.9
Torch: '2.0.1+cu117' also tried with torch 1.13 (same error)
Super-gradients: 3.1.2

During the training of yolo-nas-s i get the following error:
"'charmap' codec can't encode characters in position 20-22: character maps to <undefined>"
The training always reaches the first epoch until validation. After that I get the error mentioned above.

I'm using the following training parameters:
`from super_gradients.training.losses import PPYoloELoss
from super_gradients.training.metrics import DetectionMetrics_050
from super_gradients.training.metrics import DetectionMetrics_050_095

from super_gradients.training.models.detection_models.pp_yolo_e import PPYoloEPostPredictionCallback

train_params = {
    'silent_mode': False,
    "average_best_models":True,
    "warmup_mode": "linear_epoch_step",
    "warmup_initial_lr": 1e-6,
    "lr_warmup_epochs": 3,
    "initial_lr": 5e-4,
    "lr_mode": "cosine",
    "cosine_final_lr_ratio": 0.1,
    "optimizer": "Adam",
    "optimizer_params": {"weight_decay": 0.0001},
    "zero_weight_decay_on_bias_and_bn": True,
    "ema": True,
    "ema_params": {"decay": 0.9, "decay_type": "threshold"},
    "max_epochs": EPOCHS,
    "mixed_precision": True,
    "loss": PPYoloELoss(
        use_static_assigner=False,
        num_classes=3,
        reg_max=16
    ),
    "valid_metrics_list": [
        DetectionMetrics_050(
            score_thres=0.1,
            top_k_predictions=300,
            num_cls=3,
            normalize_targets=True,
            post_prediction_callback=PPYoloEPostPredictionCallback(
                score_threshold=0.01,
                nms_top_k=1000,
                max_predictions=300,
                nms_threshold=0.7
            )
        ),
        DetectionMetrics_050_095(
            score_thres=0.1,
            top_k_predictions=300,
            num_cls=len(dataset_params['classes']),
            normalize_targets=True,
            post_prediction_callback=PPYoloEPostPredictionCallback(
                score_threshold=0.01,
                nms_top_k=1000,
                max_predictions=300,
                nms_threshold=0.7
            )
        )
    ],
    "metric_to_watch": 'mAP@0.50:0.95'
}`

and the following training:
`
model = models.get(MODEL_ARCH, pretrained_weights="coco", num_classes=3).to(DEVICE)
CHECKPOINT_DIR = 'checkpoints'
 
trainer = Trainer(
    experiment_name='yolo_nas_s', 
    ckpt_root_dir=CHECKPOINT_DIR
)

trainer.train(
    model, 
    training_params=train_params, 
    train_loader=train_data, 
    valid_loader=val_data
)`

Can anybody help? Thank you!

**Comment by haider10shah** [comment]
I am facing the same issue on python 3.10.6. I think this has to do with how the results are getting printed to the console on the windows. It cant encode it. If I do silent_mode=True, I get rid of the issue but I run into another issue with NotImplemented error for Transforms. Any help on this would be greatly appreciated. Thank you

**Comment by BloodAxe** [comment]
Can you try with super gradients build from the master branch? I believe
this should be fixed there.

ср, 28 черв. 2023 р. о 02:10 Haider Shah ***@***.***> пише:

**Comment by haider10shah** [comment]
Any instructions on how to do that would be super helpful.

**Comment by BloodAxe** [comment]
Fixed in 3.1.2
